### DISCUSSION TITLE
laravel-debugbar Not Showing Because Blocked by laravel-csp

### DISCUSSION THREAD

**Discussion by sys-auditing** [spatie/laravel-csp#95] [Q&A]
Hi,

How to put nonce on ```laravel-debugbar``` inline script to avoid blocked by secure header ?

Thank you

**Comment by chinleung** [comment]
I've managed to inject the nonce to debugbar using:

```php
if (config('debugbar.enabled')) {
	Debugbar::getJavascriptRenderer()->setCspNonce(app('csp-nonce'));
}
```

**Reply by lightreign** [reply]
I get an "cannot call getJavascriptRenderer statically" error, might've been changed in a newer version of debugbar? 

I rewrote it to:

```php
if (config('debugbar.enabled')) {
	app('debugbar')->getJavascriptRenderer()->setCspNonce(app('csp-nonce'));
}
```
